src/components/input/index.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react'
+import TextField from '@material-ui/core/TextField'
+import startCase from 'lodash/startCase'
+
+const Input = ({
+    defaultValue = '',
+    pattern,
+    labelText,
+    helperText,
+    setValue,
+}) => {
+    const [isValid, setIsValid] = useState(true)
+    const handleChange = (e) => {
+        setValue(e.target.value)
+        setIsValid(pattern.test(e.target.value))
+    }
+
+    return (
+        <TextField
+            fullWidth
+            defaultValue={defaultValue}
+            onChange={handleChange}
+            error={!isValid}
+            {...(!isValid && { helperText })}
```

Comment:
I'm struggling a bit with the syntax here but I'm guessing it's the equivalent of something like
```
helperText={ !isValid ? helperText : '' }
```

Reply:
This conditionally adds an attribute to the args object for that component. If `!isValid` resolves to false, then the `helperText` object never added at all.

The example above would resolve in an args/props object that looks like
```
{
    defaultValue: 'some default value',
    error: false,
    helperText: ''
}
```

When what we'd really want is
```
{
    defaultValue: 'some default value',
    error: false
}
```

Reply:
Ah, now I'm kinda seeing it and how you're using the `...` operator to copy all the values and then conditionally add the `helperText`